packages/downloads/test/hub/hub-poll-download-metadata.test.ts

```diff
@@ -0,0 +1,219 @@
+import * as fetchMock from 'fetch-mock';
+import { hubPollDownloadMetadata } from "../../src/hub/hub-poll-download-metadata";
+import * as EventEmitter from 'eventemitter3';
+
+function delay (milliseconds: number) {
+  return new Promise(resolve => setTimeout(resolve, milliseconds));
+}
+
+describe("hubPollDownloadMetadata", () => {
+
+  afterEach(() => fetchMock.restore());
+
+  it('handle remote server 502 error', async done => {
+    try {
+      fetchMock.mock('http://hub.com/api/v3/datasets/abcdef0123456789abcdef0123456789_0/downloads?spatialRefId=4326&formats=csv', {
+        status: 502,
+      });
+      const mockEventEmitter = new EventEmitter();
+      spyOn(mockEventEmitter, 'emit');
+      hubPollDownloadMetadata({
+        host: 'http://hub.com',
+        datasetId: 'abcdef0123456789abcdef0123456789_0',
+        downloadId: 'test-id',
+        spatialRefId: '4326',
+        format: 'CSV',
+        eventEmitter: mockEventEmitter,
+        pollingInterval: 10
+      });
+      await delay(0);
+      expect(mockEventEmitter.emit as any).toHaveBeenCalledTimes(1);
+      expect((mockEventEmitter.emit as any).calls.first().args).toEqual([
+        'test-idPollingError', { detail: { error: new Error('Bad Gateway') } }
+      ])
+    } catch (err) {
+      expect(err).toEqual(undefined);
+    } finally {
+      done();
+    }
+  });
+
+  it('handle failed export', async done => {
+    try {
+      fetchMock.mock('http://hub.com/api/v3/datasets/abcdef0123456789abcdef0123456789_0/downloads?spatialRefId=4326&formats=csv', {
+        status: 200,
```

Comment:
consider moving large json fixtures to /tests/fixtures, exporting them as `const fooFixture = {...}`. in my experience, this makes the tests are easier to read... but at a cost of not seeing the data... so your call. Or I've also had a "base" fixture that I'd clone and then modify the parts that are important for the test so that's "in" the test

Reply:
Extracted to fixture in c735ae5.